.github/workflows/_reusable_run_tests.yml

```diff
@@ -150,73 +166,145 @@ jobs:
       # WireSystem
 
       - name: Build WireSystem
-        if: ${{ inputs.wire-ios-system || inputs.all }}
+        if: ${{ inputs.wire-system || inputs.all }}
         run: |
           echo "WireSystem has changes"
           echo "Building WireSystem..."
-          xcodebuild build-for-testing -workspace wire-ios-mono.xcworkspace -scheme WireSystem -resultBundlePath xcodebuild-wire-ios-system.xcresult -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee xcodebuild-wire-ios-system.log | bundle exec xcpretty
+          xcodebuild build-for-testing -workspace wire-ios-mono.xcworkspace -scheme WireSystemPackage -resultBundlePath xcodebuild-wire-system.xcresult -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee xcodebuild-wire-system.log | bundle exec xcpretty
 
       - name: Run Danger for WireSystem
-        if: false # ${{ inputs.wire-ios-system || inputs.all }}
+        if: false # ${{ inputs.wire-system || inputs.all }}
         env:
           DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          XCRESULT_PATH: xcodebuild-wire-ios-system.xcresult
+          XCRESULT_PATH: xcodebuild-wire-system.xcresult
         run: |
           bundle exec danger --dangerfile=Dangerfile.warning-workaround --danger_id="Run Danger for WireSystem" --verbose
 
       - name: Test WireSystem
-        if: ${{ inputs.wire-ios-system || inputs.all }}
+        if: ${{ inputs.wire-system || inputs.all }}
         run: |
           echo "Testing WireSystem..."
-          xcodebuild test -retry-tests-on-failure -workspace wire-ios-mono.xcworkspace -scheme WireSystem -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee -a xcodebuild-wire-ios-system.log | bundle exec xcpretty --report junit --output build/reports/WireSystem.junit
```

Comment:
I see in some places we are running with `-retry-tests-on-failure` and others not. How is this decided?

Reply:
This was added in order to deal with flaky tests. 
For the new (and hopefully cleaner) code in the packages this should not be necessary. Tests should either succeed or fail always, so also on a single run.